bitsandbytes/diagnostics/cuda.py

```diff
@@ -56,8 +59,8 @@ def find_cuda_libraries_in_path_list(paths_list_candidate: str) -> Iterable[Path
             except OSError:  # Assume an esoteric error trying to poke at the directory
                 pass
             for lib_pattern in CUDA_RUNTIME_LIB_PATTERNS:
-                for pth in dir.glob(lib_pattern):
-                    if pth.is_file():
+                for pth in dir.rglob(lib_pattern):
```

Comment:
I don't think changing to a recursive glob is a great idea. Some of the paths in `path_list_candidate` could be e.g. `/home/foo` for whatever reason, and then this will end up recursing through all of an user's files (or an entire filesystem).